src/components/SuggestionsList/SuggestionsList.tsx

```diff
@@ -41,97 +40,63 @@ export interface SuggestionsListProps extends NestableBaseComponentProps {
   onItemSelected(id: string | number): void;
 }
 
-interface SuggestionsListWithResultsProps extends SuggestionsListProps {
-  groupedItems: SuggestionItemGroupProps[];
-}
-
 export interface SuggestionItemGroupProps {
   title: string;
   items: SuggestionItem[];
 }
 
-const hasResults = (props: SuggestionsListProps): props is SuggestionsListWithResultsProps => {
-  return !!props.groupedItems && props.groupedItems.length > 0;
-};
-
-const withResultsClass = 'y-suggestions-list--with-results';
-const withStatusClass = 'y-suggestions-list--with-status';
-
 /**
  * A `SuggestionsList` displays a list of search results in a dropdown.
  */
 export default class SuggestionsList extends React.PureComponent<SuggestionsListProps> {
   public render() {
-    const status = this.getSearchStatus();
-    const results = this.getGroupedResults();
-
-    const classNames = ['y-suggestions-list', 'y-hc-border', this.props.className];
-    if (status) {
-      classNames.push(withStatusClass);
-    }
-    if (results) {
-      classNames.push(withResultsClass);
-    }
+    const {
+      className,
+      searchText,
+      selectedId,
+      onItemSelected,
+      isLoading,
+      loadingText,
+      noResultsText,
+      groupedItems = [],
+    } = this.props;
+    const hasResults = groupedItems.length > 0;
+    const classNames = getClassNames({ isLoading, hasResults });
 
     return (
-      <div className={join(classNames)}>
-        {results}
-        {status}
+      <div className={join(['y-suggestions-list', 'y-hc-border', className, classNames.root])}>
+        {hasResults && (
+          <ul className={classNames.resultList}>
+            {groupedItems.map((group: SuggestionItemGroupProps) => (
+              <li key={group.title}>
```

Comment:
hrmm, i know this isn't your change but i would use a vertical `<LayoutList`> here with `<LayoutListItem>`s instead of a plain 'ul' and 'li's. especially since we are probably not going to include our baseline CSS bundle in the SharePoint app, default styling for `ul` and `li` elements often have typography list styles applied instead of div-style like we do.